rfcs/0021-transaction-structure/0021-transaction-structure.md

```diff
@@ -0,0 +1,372 @@
+
+The specification to compute occupied capacity will have its RFC in the future, which is still a [draft](https://github.com/nervosnetwork/rfcs/issues/118) now.
+
+The transaction must create an output cell which occupied capacity is less than the cell capacity.
+
+```json
+occupied(cell) ≤ cell's capacity
+```
+### Code Locating
+
+The cell has two fields which type is `Script`. The CKB VM will run the `lock`  scripts of all the cells in inputs, and run the `type` scripts of all the cells in both inputs and outputs.
+
+We differentiate the terms script and code.
+
+- A script is the script structure.
+- The code is the RISC-V binary.
+- A code cell is cell which data is code.
+
+The script does not include the code directly. See the script structure below. Let's ignore the hash type `Type` and the field `args` now.
+
+![](script.png)
+
+When a CKB VM needs to run a script, it must find its code first. The fields `code_hash` and `hash_type` are used to locate the code.
+
+In CKB, the script code is compiled into RISC-V binary. The binary is stored as the data in a cell. When `hash_type` is "Data",  the script locates a cell which data hash equals the script's `code_hash`. The cell data hash, as the name suggests, is computed from the cell data (see Appendix A). The scope is limited in the transaction, script can only find a matched cell from `cell_deps`.
+
+![](cell-deps.png)
+
+The following diagram shows how CKB finds a matched script code.
+
+![](code-locating.png)
+
+If you want to use a script in CKB, follow the code locating rules:
+
+- Compile your code into RISC-V binary. You can find some examples in the [repository](https://github.com/nervosnetwork/ckb-system-scripts) which builds the code for system cells.
+- Create a cell which stores the binary as data in a transaction, and send the transaction to the chain.
+- Construct a script structure, which `hash_type` is "Data", and `code_hash` is just the hash of the built binary.
+- Use the script as the type or the lock script in a cell.
+- If the script has to run in a transaction, include the code cell out point in the `cell_deps`.
+
+The cells in `cell_deps` must be live, just like `inputs`. Unlike `inputs`, a cell only used in `cell_deps` is not considered dead.
+
+The following two chapters will talk about how the script is used in a transaction to lock the cells and establish contracts on cells.
+
+### Lock Script
+
+Every cell has a lock script. The lock script must run when the cell is used as an input in a transaction. When the script only appears in the outputs, it is not required to reveal the corresponding code in `cell_deps`. A transaction is valid only when all the lock scripts in the inputs exit normally. Since the script runs on inputs, it acts as the lock to controls who can unlock and destroy the cell, as well as spend the capacity stored in the cell.
```

Comment:
```suggestion
Every cell has a lock script. The lock script must run when the cell is used as an input in a transaction. When the script only appears in the outputs, it is not required to reveal the corresponding code in `cell_deps`. A transaction is valid only when all the lock scripts in the inputs exit normally. Since the script runs on inputs, it acts as the lock to control who can unlock and destroy the cell, as well as spend the capacity stored in the cell.
```